pkg/pillar/types/dpc.go

```diff
@@ -401,6 +402,27 @@ func (config *DevicePortConfig) DoSanitize(log *base.LogObject, args DPCSanitize
 				config.Ports[i].IsL3Port = true
 			}
 		}
+
+		sanitizeNTPServers := func(servers *[]string) []string {
+			var sanitizedNTPServers []string
+			if *servers == nil {
+				return nil
+			}
+			for _, ntpServer := range *servers {
+				if strings.TrimSpace(ntpServer) != "" {
+					sanitizedNTPServers = append(sanitizedNTPServers, ntpServer)
```

Comment:
Wait - don't you want to append the trimmed string to the result?
Or are you only concerned with filtering out ntpServer which contains only whitspace?